vue-server/vite.config.ts

```diff
@@ -98,12 +101,17 @@ function vitePluginVueServer(): PluginOption {
 }
 
 function patchServerVue(plugin: Plugin): Plugin {
+	// need to force non-ssr transform to always render vnode
 	tinyassert(typeof plugin.transform === "function");
 	const oldTransform = plugin.transform;
 	plugin.transform = async function (code, id, _options) {
-		// need to force non-ssr transform to always render vnode
 		return oldTransform.apply(this, [code, id, { ssr: false }]);
 	};
+	tinyassert(typeof plugin.load === "function");
+	const oldLoad = plugin.load;
+	plugin.load = async function (id, _options) {
+		return oldLoad.apply(this, [id, { ssr: false }]);
+	};
```

Comment:
It looks like we also need to patch `load` since it picks up cache based on `ssr` flag https://github.com/vitejs/vite-plugin-vue/blob/46d0baa45c9e7cf4cd3ed773af5ba9f2a503b446/packages/plugin-vue/src/index.ts#L229-L230